app/routes/authentication.py

```diff
@@ -154,3 +171,27 @@ def confirm_user(token):
     db.session.add(user)
     db.session.flush()
     return redirect("https://holepunch.io/email_confirmed/")
+
+
+@auth_blueprint.route("/confirm_password_reset/<token>", methods=["POST"])
```

Comment:
I'm not sure it's necessary to have multiple URLS handling token confirmation.  You can think of the token as a temporary, one-time-password for logging in so that a user can reset their password.   Basically the flow would be:

1) Request Password Reset (GET /reset_password or GET /user/1/reset_password)
2) We send an email with the token in it.
3) Visiting the url with that token in it provides a one time log-in to the user admin area.
4) You can then `PUT/PATCH` to `/user/1` to reset the password.

It's also possible that I'm misunderstanding what you're doing here and that you have a different flow in mind.

Reply:
I agree we can have one endpoint to check tokens. To do that we need to check the tokens based on the "salt" so we know what to do with the token(confirm email, change password, etc..) Also to allow a one time login we need to implement a jwt blacklist and not return a refresh token